custom_components/bond/fan.py

```diff
@@ -86,10 +97,12 @@ def turn_on(self, speed=None, **kwargs):
 
     def turn_off(self, **kwargs):
         """Instruct the fan to turn off"""
+        self._attributes['current_speed'] = SPEED_OFF
         self._bond.turnOff(self._deviceId)
 
     def set_speed(self, speed: str) -> None:
         """Set the speed of the fan."""
+        self._attributes['current_speed'] = speed
```

Comment:
I'd set this after the call to our bond api. This way if the call fails the current reported speed will be correct and we wouldn't have to wait for the next call to update for it to get corrected.

Reply:
Great catch - I've updated the PR